include/aws/common/thread.h

```diff
@@ -237,6 +238,14 @@ AWS_COMMON_API void aws_thread_increment_unjoined_count(void);
  */
 AWS_COMMON_API void aws_thread_decrement_unjoined_count(void);
 
+/**
+ * Gets name of the current thread.
```

Comment:
While we're doing this, let's have the function take a `aws_thread_id_t` instead of only being for the current thread.

That's the API for every lower-level get-thread-name function we're calling into

You could still offer the _current_name function as a convenience thing, or not

Reply:
Im not opposed to expose thread id version as well, but there are a couple reasons i decided not to add it to pr:
1. we currently dont have apis that take thread id, but we have couple apis that work on current thread. use case for this is to get thread name in the callback, so i didnt want to overcomplicate
2. on windows we would need to open/close handle to use thread id. minor thing to implement, but seemed like overkill